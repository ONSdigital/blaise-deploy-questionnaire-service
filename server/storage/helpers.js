import {getEnvironmentVariables} from "../Config";

const util = require("util");
const gc = require("./config");

const {BUCKET_NAME} = getEnvironmentVariables();
const bucket = gc.bucket(BUCKET_NAME);

const {format} = util;

/**
 *
 * @param { File } object file object that will be uploaded
 * @description - This function does the following
 * - It uploads a file to the image bucket on Google Cloud
 * - It accepts an object as an argument with the
 *   "originalname" and "buffer" as keys
 */

const uploadImage = (file) => new Promise((resolve, reject) => {
    const {name} = file;

    const blob = bucket.file(name.replace(/ /g, "_"));
    const blobStream = blob.createWriteStream({
        resumable: false
    });

    console.log(`Bucket ${bucket.name}`);

    blobStream.on("finish", () => {
        console.log("upload finished ");
        const publicUrl = format(
            `https://storage.googleapis.com/${bucket.name}/${blob.name}`
        );
        resolve(publicUrl);
    }).on("error", (error) => {
        console.log("blobStream failed");
        reject("Unable to upload image, something went wrong");
    }).end(file.getContent());

});

export const checkFile = (filename) => new Promise((resolve, reject) => {
    async function getMetadata() {
        // Gets the metadata for the file
        const [metadata] = await bucket.file(filename).getMetadata();

        const file = {
            name: metadata.name,
            updated: metadata.updated,
            found: true
        };
        return file;
    }

    getMetadata()
        .then((file) => {
            resolve(file);
        }).catch((error) => {
        console.log(error.code);
        if (error.code === 404) {
            resolve({ found: false});
        }
        reject(`Failed ${error}`);
    });
});

module.exports = {uploadImage, checkFile};
