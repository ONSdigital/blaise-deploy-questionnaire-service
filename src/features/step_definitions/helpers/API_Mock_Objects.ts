import { Instrument } from "../../../../Interfaces";

export const instrumentList: Instrument[] = [{
    name: "OPN2101A",
    serverParkName: "gusty",
    installDate: "2021-01-15T14:41:29.4399898+00:00",
    status: "Active",
    dataRecordCount: 0,
    hasData: false,
    active: false,
    fieldPeriod: "January 2021"
}, {
    name: "OPN2007T",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:18:40.1503617+00:00",
    status: "Active",
    dataRecordCount: 10,
    hasData: true,
    active: true,
    fieldPeriod: "July 2020"
}, {
    name: "OPN2004A",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:26:43.4233454+00:00",
    status: "Failed",
    dataRecordCount: 0,
    hasData: false,
    active: false,
    fieldPeriod: "April 2020"
}];

export const opnInstrument: Instrument = {
    name: "OPN2004A",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:26:43.4233454+00:00",
    status: "Active",
    dataRecordCount: 0,
    hasData: false,
    active: false,
    fieldPeriod: "April 2020"
};

export const instrumentWithName = (instrumentName: string): Instrument => {
    return {
        name: instrumentName,
        serverParkName: "gusty",
        installDate: "2021-01-15T15:26:43.4233454+00:00",
        status: "Active",
        dataRecordCount: 0,
        hasData: false,
        active: false,
        fieldPeriod: "April 2020"
    };
};
