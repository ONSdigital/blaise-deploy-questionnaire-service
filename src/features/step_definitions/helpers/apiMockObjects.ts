import { AxiosError, AxiosHeaders } from "axios";
import { Questionnaire } from "blaise-api-node-client";

export const questionnaireList: Questionnaire[] = [{
    name: "OPN2101A",
    serverParkName: "gusty",
    installDate: "2021-01-15T14:41:29.4399898+00:00",
    status: "Active",
    dataRecordCount: 0,
    hasData: false,
    active: false,
    fieldPeriod: "January 2021",
    blaiseVersion: "5.9.9.2735"
}, {
    name: "OPN2007T",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:18:40.1503617+00:00",
    status: "Active",
    dataRecordCount: 10,
    hasData: true,
    active: true,
    fieldPeriod: "July 2020",
    blaiseVersion: "5.9.9.2735"

}, {
    name: "OPN2004A",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:26:43.4233454+00:00",
    status: "Failed",
    dataRecordCount: 0,
    hasData: false,
    active: false,
    fieldPeriod: "April 2020",
    blaiseVersion: "5.9.9.2735"

}];

export const opnQuestionnaire: Questionnaire = {
    name: "OPN2004A",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:26:43.4233454+00:00",
    status: "Active",
    dataRecordCount: 0,
    hasData: false,
    active: false,
    fieldPeriod: "April 2020",
    blaiseVersion: "5.9.9.2735"
};

export const ipsQuestionnaire: Questionnaire = {
    name: "IPS1337a",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:26:43.4233454+00:00"
};

export const ipsPilotQuestionnaire: Questionnaire = {
    name: "IPS2500a",
    serverParkName: "gusty",
    installDate: "2022-01-15T15:26:43.4233454+00:00"
};

export const questionnaireWithName = (questionnaireName: string): Questionnaire => {
    return {
        name: questionnaireName,
        serverParkName: "gusty",
        installDate: "2021-01-15T15:26:43.4233454+00:00",
        status: "Active",
        dataRecordCount: 0,
        hasData: false,
        active: false,
        fieldPeriod: "April 2020",
        blaiseVersion: "5.9.9.2735"
    };
};

export const cloudFunctionAxiosError: AxiosError = new AxiosError();
cloudFunctionAxiosError.response = {
    status: 500,
    statusText: "Internal Server Error",
    data: "Error creating IPS donor cases: No users found with role 'IPS Manager'",
    headers: {},
    config: {
        headers: new AxiosHeaders(),
    },
};

export const mockSuccessResponseForDonorCasesCreation = {
    data: "Success",
    status: 200,
};