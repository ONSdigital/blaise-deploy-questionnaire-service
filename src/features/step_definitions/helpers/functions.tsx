import { createMemoryHistory } from "history";
import { act, render, screen } from "@testing-library/react";
import { Router } from "react-router";
import App from "../../../app";
import flushPromises from "../../../tests/utils";
import React from "react";
import userEvent from "@testing-library/user-event";

/*
* Renders the App in a Mock Router, then navigates the the 'Deploy a questionnaire' page
* and then selects a mock OPN2004A.bpkg in the File select Input field.
*  */
export async function navigateToDeployPageAndSelectFile(questionnaire = "OPN2004A"): Promise<void> {
    const history = createMemoryHistory();
    render(
        <Router history={history}>
            <App />
        </Router>
    );
    await act(async () => {
        await flushPromises();
    });

    userEvent.click(screen.getByText(/Deploy a questionnaire/i));

    const input = screen.getByLabelText(/Select survey package/i);

    const file = new File(["(⌐□_□)"], `${questionnaire}.bpkg`, { type: "application/zip" });

    userEvent.upload(input, file);
}

export default async function navigateToDeployPageAndSelectFileAndContinue(questionnaire = "OPN2004A"): Promise<void> {
    await navigateToDeployPageAndSelectFile(questionnaire)
    await clickContinue();
}

export async function navigatePastSettingTOStartDateAndDeployQuestionnaire(): Promise<void> {
    await selectNoTOStartDateAndContinue();
    await clickContinue();
    await clickDeployQuestionnaire();
}

export async function navigateToSetTMReleaseDatePageAndContinue(questionnaire: string): Promise<void> {
    await navigateToDeployPageAndSelectFile(questionnaire);
    await clickContinue();

    // TO start date page
    await selectNoTOStartDateAndContinue();
    await clickContinue();
}

export async function selectNoTOStartDateAndContinue(): Promise<void> {
    userEvent.click(screen.getByText(/No start date/i));
}

export async function selectNoTMReleaseDateAndContinue(): Promise<void> {
    userEvent.click(screen.getByText(/No release date/i));
    userEvent.click(screen.getByText(/Continue/));
    await act(async () => {
        await flushPromises();
    });
}

export async function clickDeployQuestionnaire(): Promise<void> {
    userEvent.click(screen.getByText(/Deploy questionnaire/));
    await act(async () => {
        await flushPromises();
    });
}

export async function clickContinue(): Promise<void> {
    userEvent.click(screen.getByText(/Continue/));
    await act(async () => {
        await flushPromises();
    });
}

export function formatDateString(date: string): string {
    const splitDate = date.split("/");
    return `${splitDate[2]}-${splitDate[1]}-${splitDate[0]}`;
}
