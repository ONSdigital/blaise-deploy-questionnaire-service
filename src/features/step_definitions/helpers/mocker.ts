export type RouteMock = {
  Path: string;
  Method?: string;
  Status: number;
  JSON?: any;
}

export class Mocker {
  mocks: RouteMock[];

  constructor() {
    this.mocks = [];
  }

  set(new_mock: RouteMock): void {
    this.mocks.forEach((mock, index) => {
      if (mock.Path == new_mock.Path && mock.Method == new_mock.Method) {
        this.mocks.splice(index, 1);
      }
    });
    this.mocks.push(new_mock);
  }

  mocker(url: RequestInfo, config?: RequestInit): Promise<Partial<Response>> {
    let mock = this.exactMatch(url, config);
    if (mock === undefined) {
      mock = this.exactPathMatch(url);
    }
    if (mock === undefined) {
      mock = this.partialMatchWithMethod(url, config);
    }
    if (mock === undefined) {
      mock = this.partialMatch(url);
    }
    if (mock) {
      console.log("Matched mock");
      console.log(mock);
      let json = mock.JSON;
      if (json === undefined) {
        json = {};
      }
      return Promise.resolve({
        status: mock.Status,
        json: () => Promise.resolve(json)
      });
    }
    console.log("No matching mock");
    console.log(url);
    return Promise.reject("No matching mock");
  }

  applyMocks(): void {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    global.fetch = jest.fn((url: RequestInfo, config?: RequestInit) => this.mocker(url, config));
  }

  exactMatch(url: RequestInfo, config?: RequestInit): RouteMock | undefined {
    for (const mock of this.mocks) {
      if (mock.Path === url.toString() && config?.method == mock.Method) {
        return mock;
      }
    }
    return undefined;
  }

  exactPathMatch(url: RequestInfo): RouteMock | undefined {
    for (const mock of this.mocks) {
      if (mock.Path === url.toString() && mock.Method === undefined) {
        return mock;
      }
    }
    return undefined;
  }

  partialMatchWithMethod(url: RequestInfo, config?: RequestInit): RouteMock | undefined {
    for (const mock of this.mocks) {
      if (url.toString().includes(mock.Path) && config?.method === mock.Method) {
        return mock;
      }
    }
    return undefined;
  }

  partialMatch(url: RequestInfo,): RouteMock | undefined {
    for (const mock of this.mocks) {
      if (url.toString().includes(mock.Path) && mock.Method === undefined) {
        return mock;
      }
    }
    return undefined;
  }
}
