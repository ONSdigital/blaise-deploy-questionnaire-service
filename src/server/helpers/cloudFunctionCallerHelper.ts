import axios from "axios";
import { GoogleAuth } from "google-auth-library";

export async function getIdTokenFromMetadataServer(targetAudience: string) {
    const googleAuth = new GoogleAuth();

    const client = await googleAuth.getIdTokenClient(targetAudience);

    return await client.idTokenProvider.fetchIdToken(targetAudience);
}

export async function callCloudFunction(url: string, payload: any): Promise<{ message: string, status: number }> {

    const token = await getIdTokenFromMetadataServer(url);

    try {
        const res = await axios.post(url, payload, {
            headers: {
                "Content-Type": "application/json",
                Authorization: `Bearer ${token}`,
            },
        });

        return {
            message: res.data,
            status: res.status
        };
    } catch (error) {
        console.error("Error:", error);
        return {
            message: (error as any).response.data,
            status: 500
        };
    }
}
