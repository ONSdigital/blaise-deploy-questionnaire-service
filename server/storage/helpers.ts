import {getEnvironmentVariables} from "../Config";

import {storage} from "./config";
import {GetSignedUrlConfig} from "@google-cloud/storage";

const {BUCKET_NAME} = getEnvironmentVariables();
const bucket = storage.bucket(BUCKET_NAME);

const getSignedUrl = (filename: string): Promise<string> => new Promise((resolve, reject) => {
    async function getSignedUrl() {

        const options = <GetSignedUrlConfig>{
            version: "v4",
            action: "write",
            expires: Date.now() + 15 * 60 * 1000, // 15 minutes
            contentType: "application/octet-stream",
        };

        // Get a v4 signed URL for uploading file
        const [url] = await bucket
            .file(filename)
            .getSignedUrl(options);
        return url;
    }

    getSignedUrl()
        .then((url) => {
            resolve(url);
        }).catch((error) => {
        console.error(error, "getSignedUrl Failed");
        reject("getSignedUrl Failed");
    });
});

const getBucketItems = (): Promise<string[]> => new Promise((resolve, reject) => {
    async function getBucketItems() {
        const [files] = await bucket.getFiles();
        const fileList: string[] = [];

        files.map(({name}) => {
            if (name.endsWith(".bpkg")) {
                fileList.push(name);
            }
        });
        return fileList;
    }

    getBucketItems()
        .then((files) => {
            resolve(files);
        }).catch((error) => {
        console.error(error, "getBucketItems Failed");
        reject("getBucketItems Failed");
    });
});

interface file {
    name?: string
    updated?: string
    found: boolean
}

const checkFile = (filename: string): Promise<file> => new Promise((resolve, reject) => {
    async function getMetadata() {
        // Gets the metadata for the file
        const [metadata] = await bucket.file(filename).getMetadata();

        return {
            name: metadata.name,
            updated: metadata.updated,
            found: true
        };
    }

    getMetadata()
        .then((file) => {
            resolve(file);
        }).catch((error) => {
        console.log(error.code);
        if (error.code === 404) {
            resolve({found: false});
        }
        reject(`Failed ${error}`);
    });
});


export {checkFile, getSignedUrl, getBucketItems};
