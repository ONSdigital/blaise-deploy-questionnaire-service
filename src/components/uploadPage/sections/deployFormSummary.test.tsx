/**
 * @jest-environment jsdom
 */

import {render} from "@testing-library/react";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import '@testing-library/jest-dom/extend-expect';

const mock = new MockAdapter(axios);

import { AuthManager } from "blaise-login-react-client";
import React from "react";
import AskToSetTMReleaseDate from "./askToSetTMReleaseDate";
import { Formik } from "formik";
import DeployFormSummary from "./deployFormSummary";
import {questionnaireWithName} from "../../../features/step_definitions/helpers/apiMockObjects";

jest.mock("blaise-login-react-client");
AuthManager.prototype.loggedIn = jest.fn().mockImplementation(() => {
    return Promise.resolve(true);
});

describe("Deploy form summary page", () => {
    const lmsQuestionnaireName = "LMS2004A"
    const opnQuestionnaireName = "OPN2004A"

    const lmsFile = new File(["龴ↀ◡ↀ龴"], `${lmsQuestionnaireName}.bpkg`, { type: "application/pdf" });
    const opnFile = new File(["(♥_♥)"], `${opnQuestionnaireName}.bpkg`, { type: "application/pdf" });

    const lmsQuestionnaire = questionnaireWithName(lmsQuestionnaireName);
    const opnQuestionnaire = questionnaireWithName(opnQuestionnaireName);

    it("should match the Snapshot", async () => {
        const wrapper = render(
            <Formik initialValues={{ }} onSubmit={( ) => { }} >
                <DeployFormSummary  file={lmsFile} foundQuestionnaire={lmsQuestionnaire}/>
            </Formik>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it("should display the questionnaire file name", async () => {
        const { getByText } = render(
            <Formik initialValues={{ }} onSubmit={( ) => { }} >
                <DeployFormSummary  file={lmsFile} foundQuestionnaire={lmsQuestionnaire}/>
            </Formik>
        );

        expect(getByText(/Questionnaire file name/i)).toBeDefined();
    });

    it("should display when the file was last modified", async () => {
        const { getByText } = render(
            <Formik initialValues={{ }} onSubmit={( ) => { }} >
                <DeployFormSummary  file={lmsFile} foundQuestionnaire={lmsQuestionnaire}/>
            </Formik>
        );

        expect(getByText(/Questionnaire file last modified date/i)).toBeDefined();
    });

    it("should display the questionnaire file size", async () => {
        const { getByText } = render(
            <Formik initialValues={{ }} onSubmit={( ) => { }} >
                <DeployFormSummary  file={lmsFile} foundQuestionnaire={lmsQuestionnaire}/>
            </Formik>
        );

        expect(getByText(/Questionnaire file size/i)).toBeDefined();
    });

    it("should display if the questionnaire exists in Blaise", async () => {
        const { getByText } = render(
            <Formik initialValues={{ }} onSubmit={( ) => { }} >
                <DeployFormSummary  file={lmsFile} foundQuestionnaire={lmsQuestionnaire}/>
            </Formik>
        );

        expect(getByText(/Does the questionnaire already exist in blaise?/i)).toBeDefined();
    });

    it("should display the telephone operation start date", async () => {
        const { getByText } = render(
            <Formik initialValues={{ }} onSubmit={( ) => { }} >
                <DeployFormSummary  file={lmsFile} foundQuestionnaire={lmsQuestionnaire}/>
            </Formik>
        );

        expect(getByText(/Set a telephone operations start date for questionnaire?/i)).toBeDefined();
    });

    it("should display the totalmobile release date for LMS questionnaires", async () => {
        const { getByText } = render(
            <Formik initialValues={{ }} onSubmit={( ) => { }} >
                <DeployFormSummary  file={lmsFile} foundQuestionnaire={lmsQuestionnaire}/>
            </Formik>
        );

        expect(getByText(/Set a totalmobile release date for questionnaire?/i)).toBeDefined();
    });

    it("should not display the totalmobile release date for non-LMS questionnaires", async () => {
        const { queryByText } = render(
            <Formik initialValues={{ }} onSubmit={( ) => { }} >
                <DeployFormSummary  file={opnFile} foundQuestionnaire={opnQuestionnaire}/>
            </Formik>
        );

        expect(queryByText(/Set a totalmobile release date for questionnaire?/i)).not.toBeInTheDocument();
    });
});
