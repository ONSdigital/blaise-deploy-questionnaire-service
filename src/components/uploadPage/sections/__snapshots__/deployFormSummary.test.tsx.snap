// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deploy form summary page should match the Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="summary"
      >
        <div
          class="summary__group"
        >
          <h1
            class="summary__group-title"
          >
            Deployment summary
          </h1>
          <table
            class="summary__items"
          >
            <thead
              class="u-vh"
            >
              <tr>
                <th>
                  Question
                </th>
                <th>
                  Answer given
                </th>
              </tr>
            </thead>
            <tbody
              class="summary__item"
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    Questionnaire file name
                  </div>
                </td>
                <td
                  class="summary__values"
                >
                  LMS2004A.bpkg
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item"
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    Questionnaire file last modified date
                  </div>
                </td>
                <td
                  class="summary__values"
                >
                  24/01/2022 01:02
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item"
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    Questionnaire file size
                  </div>
                </td>
                <td
                  class="summary__values"
                >
                  1
                  MB
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item"
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    Does the questionnaire already exist in blaise?
                  </div>
                </td>
                <td
                  class="summary__values"
                >
                  Yes, overriding questionnaire
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item"
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    Set a telephone operations start date for questionnaire?
                  </div>
                </td>
                <td
                  class="summary__values"
                >
                  Start date not specified
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item"
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    Set a totalmobile release date for questionnaire?
                  </div>
                </td>
                <td
                  class="summary__values"
                >
                  Release date not specified
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="summary"
    >
      <div
        class="summary__group"
      >
        <h1
          class="summary__group-title"
        >
          Deployment summary
        </h1>
        <table
          class="summary__items"
        >
          <thead
            class="u-vh"
          >
            <tr>
              <th>
                Question
              </th>
              <th>
                Answer given
              </th>
            </tr>
          </thead>
          <tbody
            class="summary__item"
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  Questionnaire file name
                </div>
              </td>
              <td
                class="summary__values"
              >
                LMS2004A.bpkg
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item"
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  Questionnaire file last modified date
                </div>
              </td>
              <td
                class="summary__values"
              >
                24/01/2022 01:02
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item"
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  Questionnaire file size
                </div>
              </td>
              <td
                class="summary__values"
              >
                1
                MB
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item"
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  Does the questionnaire already exist in blaise?
                </div>
              </td>
              <td
                class="summary__values"
              >
                Yes, overriding questionnaire
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item"
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  Set a telephone operations start date for questionnaire?
                </div>
              </td>
              <td
                class="summary__values"
              >
                Start date not specified
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item"
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  Set a totalmobile release date for questionnaire?
                </div>
              </td>
              <td
                class="summary__values"
              >
                Release date not specified
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
